File: versions/forge1710/src/main/scala/com/idkidknow/mcreallink/forge1710/LifecycleEvents.scala
package com.idkidknow.mcreallink.forge1710

import net.minecraft.server.MinecraftServer

@SuppressWarnings(Array("org.wartremover.warts.Var"))
object LifecycleEvents {
  var serverStartingCallback: MinecraftServer => Unit = { _ => () }
  var serverStoppingCallback: () => Unit = { () => () }
}

File: versions/forge1122/src/common/java/com/idkidknow/mcreallink/forge1122/events/LifecycleEvents.java
package com.idkidknow.mcreallink.forge1122.events;

import net.minecraft.server.MinecraftServer;

import java.util.function.Consumer;

public class LifecycleEvents {
    private static Consumer<MinecraftServer> serverStartingCallback;
    private static Runnable serverStoppingCallback;

    public static Consumer<MinecraftServer> getServerStartingCallback() {
        return serverStartingCallback;
    }

    public static Runnable getServerStoppingCallback() {
        return serverStoppingCallback;
    }

    public static void setServerStartingCallback(Consumer<MinecraftServer> serverStartingCallback) {
        LifecycleEvents.serverStartingCallback = serverStartingCallback;
    }

    public static void setServerStoppingCallback(Runnable serverStoppingCallback) {
        LifecycleEvents.serverStoppingCallback = serverStoppingCallback;
    }
}
